Source/RealmResultsController.swift

```diff
@@ -175,16 +190,16 @@ public class RealmResultsController<T: Object, U> : RealmResultsCacheDelegate {
         return self.mapper(object)
     }
 
-    private func keyPathIsValid(keyPath: String?, sorts: [SortDescriptor]) -> Bool {
-        if keyPath == nil { return true }
-        guard let firstSort = sorts.first else { return false }
-        return keyPath == firstSort.property
-    }
-    
     private func sortDescriptorsAreEmpty(sorts: [SortDescriptor]) -> Bool {
         return sorts.first == nil
     }
     
+    // At this point, we are sure sorts.first always has a SortDescriptor
+    private func keyPathIsValid(keyPath: String?, sorts: [SortDescriptor]) -> Bool {
+        if keyPath == nil { return true }
+        return keyPath == sorts.first!.property
```

Comment:
yes, this method won't be called otherwise. Doesn't need to add extra defensive because then it's impossible to test the fallback :P